File: Day<24>/Solution.cpp
#include <bits/stdc++.h>
using namespace std;


 // } Driver Code Ends
class Solution{
    public:
    vector<int> TopK(vector<int>& array, int k)
    {
        int i,j=0;
        int arr[10002];
        for(i=0;i<10002;i++)
            arr[i] = 0;
            
        for(i=0;i<array.size();i++)
            arr[array[i]]++;
        
        vector<int>res[100002];
        
        for(i=0;i<10002;i++)
            {
                if(arr[i]!=0)
                    res[arr[i]].push_back(i);
            }
            
        vector<int>ans;
        for(i=100001;i>=0;i--)
            {
                for(int p = res[i].size()-1;p>=0;p--)
                    {
                        if(j!=k)
                            ans.push_back(res[i][p]);
                        else
                            break;
                        j++;
                    }
            }
        return ans;
    }
};

// { Driver Code Starts.

int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        int n,k ;
        cin>>n;
        
        vector<int>array(n);
        
        for(int i=0; i<n; i++)
            cin>>array[i];
            
        cin>>k;
        
        Solution obj;
        vector<int> result = obj.TopK(array,k);
        
        for(int i=0; i<result.size();i++)
            cout<<result[i]<<" ";

        cout<<"\n";

    }
    return 0;
}

File: Day<24>/Solution.java
import java.io.*;
import java.util.*;

class GfG
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-->0)
        {
            int n = sc.nextInt();
            ArrayList<Integer> arr = new ArrayList<Integer>();
            ArrayList<Integer> res = new ArrayList<Integer>();
            for (int i=0; i<n; i++)
            {
                int x = sc.nextInt();
                arr.add(x);
            }
            int k = sc.nextInt();
            
            Solution obj = new Solution();
            res = obj.TopK(arr,k);
            
            for (int i=0; i<res.size(); i++)
            {
                System.out.print(res.get(i) + " ");
            }
            System.out.println();
        }
    }
}

class Solution
{
    ArrayList<Integer>TopK(ArrayList<Integer> array, int k)
    {
        int i,j=0;
        int arr[] = new int[10002];
        for (i=0; i<10002; i++)
            arr[i] = 0;
            
        for (i=0; i<array.size(); i++)
            arr[array.get(i)]++;
        
        ArrayList<Integer>res[] = new ArrayList[100002];
        for (i=0; i<100002; i++)
        {
            ArrayList<Integer>temp = new ArrayList<Integer>();
            res[i]=temp;
        }
        for (i=0; i<10002; i++)
            if (arr[i]!=0)
                res[arr[i]].add(i);

        ArrayList<Integer>ans = new ArrayList<Integer>();
        for (i=100001; 0<=i; i--)
        {
            for (int p=res[i].size()-1; 0<=p; p--)
            {
                if (j!=k)
                    ans.add(res[i].get(p));
                else
                    break;
                j++;
            }
        }
        return ans;
    }
}
